sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingQuerySuite.scala

```diff
@@ -460,9 +460,9 @@ class StreamingQuerySuite extends StreamTest with BeforeAndAfter with Logging wi
     val streamingInputDF = createSingleTriggerStreamingDF(streamingTriggerDF).toDF("value")
 
     val progress = getFirstProgress(streamingInputDF.join(streamingInputDF, "value"))
-    assert(progress.numInputRows === 20) // data is read multiple times in self-joins
```

Comment:
So IIUC in this line, the EXCHANGE_REUSE_ENABLED == true, and its not read twice actually?

Reply:
The exchange reuse is not triggered here, because the project of one side is eliminated. In the kafka test, we have a cast in the project so Spark doesn't eliminate project ay any side, and the trigger exchange reuse.